shared/src/types/ethereum_events.rs

```diff
@@ -1,214 +1,321 @@
-    /// events will be unique.
-    pub nonce: Nonce,
+pub struct KeccakHash(pub [u8; 32]);
+
+/// An Ethereum event to be processed by the Anoma ledger
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, BorshSchema)]
+pub enum EthereumEvent {
+    /// Event transferring batches of ether or Ethereum based ERC20 tokens
+    /// from Ethereum to wrapped assets on Anoma
+    TransfersToNamada {
+        /// Monotonically increasing nonce
+        #[allow(dead_code)]
+        nonce: Uint,
+        /// The batch of transfers
+        #[allow(dead_code)]
+        transfers: Vec<TransferToNamada>,
+    },
+    /// A confirmation event that a batch of transfers have been made
+    /// from Anoma to Ethereum
+    TransfersToEthereum {
+        /// Monotonically increasing nonce
+        #[allow(dead_code)]
+        nonce: Uint,
+        /// The batch of transfers
+        #[allow(dead_code)]
+        transfers: Vec<TransferToEthereum>,
+    },
+    /// Event indication that the validator set has been updated
+    /// in the governance contract
+    ValidatorSetUpdate {
+        /// Monotonically increasing nonce
+        #[allow(dead_code)]
+        nonce: Uint,
+        /// Hash of the validators in the bridge contract
+        #[allow(dead_code)]
+        bridge_validator_hash: KeccakHash,
+        /// Hash of the validators in the governance contract
+        #[allow(dead_code)]
+        governance_validator_hash: KeccakHash,
+    },
+    /// Event indication that a new smart contract has been
+    /// deployed
+    NewContract {
+        /// Name of the contract
+        #[allow(dead_code)]
+        name: String,
+        /// Address of the contract on Ethereum
+        #[allow(dead_code)]
+        address: EthAddress,
+    },
+    /// Event indicating that a smart contract has been updated
+    UpgradedContract {
+        /// Name of the contract
+        #[allow(dead_code)]
+        name: String,
+        /// Address of the contract on Ethereum
+        #[allow(dead_code)]
+        address: EthAddress,
+    },
+    /// Event indication a new Ethereum based token has been whitelisted for
+    /// transfer across the bridge
+    UpdateBridgeWhitelist {
+        /// Monotonically increasing nonce
+        #[allow(dead_code)]
+        nonce: Uint,
+        /// Tokens to be allowed to be transferred across the bridge
+        #[allow(dead_code)]
+        whitelist: Vec<TokenWhitelist>,
+    },
 }
 
 impl EthereumEvent {
     /// SHA256 of the Borsh serialization of the [`EthereumEvent`].
+    #[allow(dead_code)]
     fn hash(&self) -> Result<Hash, std::io::Error> {
         let bytes = self.try_to_vec()?;
         Ok(Hash::sha256(&bytes))
     }
 }
 
-/// Representation of address on Ethereum
+/// An event transferring some kind of value from Ethereum to Anoma
 #[derive(
-    Clone, PartialEq, Eq, Debug, BorshSerialize, BorshDeserialize, BorshSchema,
+    Clone, Debug, PartialEq, BorshSerialize, BorshDeserialize, BorshSchema,
```

Comment:
Should we derive `Eq` as well as `PartialEq` if we can do so? I am running a later nightly and see this clippy warning for the anoma codebase often

```
warning: you are deriving `PartialEq` and can implement `Eq`
  --> shared/src/types/validity_predicate.rs:11:5
   |
11 |     PartialEq,
   |     ^^^^^^^^^ help: consider deriving `Eq` as well: `PartialEq, Eq`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derive_partial_eq_without_eq
```

Reply:
Sure, we can do that